cadc-cdp/src/main/java/ca/nrc/cadc/cred/CertUtil.java

```diff
@@ -129,140 +144,119 @@ public class CertUtil
      * @throws CertificateNotYetValidException
      * @throws CertificateExpiredException
      */
-    public static X509Certificate generateCertificate(PKCS10CertificationRequest csr, 
-            int lifetime, X509CertificateChain chain) 
-        throws NoSuchAlgorithmException, NoSuchProviderException, InvalidKeyException,
+    public static X509Certificate generateCertificate(PKCS10CertificationRequest csr,
+            int lifetime, X509CertificateChain chain)
+            throws NoSuchAlgorithmException, NoSuchProviderException, InvalidKeyException,
             CertificateParsingException, CertificateEncodingException,
             SignatureException, CertificateExpiredException,
-            CertificateNotYetValidException
-    {
-        X509Certificate issuerCert = chain.getChain()[0];
-        PrivateKey issuerKey = chain.getPrivateKey();
+            CertificateNotYetValidException {
 
+        X509Certificate issuerCert = chain.getChain()[0];
+        final PrivateKey issuerKey = chain.getPrivateKey();
         Security.addProvider(new BouncyCastleProvider());
-
-        X509V3CertificateGenerator certGen = new X509V3CertificateGenerator();
-
-        certGen.setSerialNumber(BigInteger.valueOf(System
-                .currentTimeMillis()));
-        certGen.setIssuerDN(issuerCert.getSubjectX500Principal());
-
-        // generate the proxy DN as the issuerDN + CN=random number
+        
+        final BigInteger serial = BigInteger.valueOf(System.currentTimeMillis());
```

Comment:
Can you move this closer to where it's needed? I'm surprised that `checkstyle` didn't complain about this.

Reply:
moved; checkstyle allows it when `final`

Reply:
Good to know the `final` detail.